shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -279,66 +279,60 @@ private void setSystemChromeEnabledSystemUIMode(PlatformChannel.SystemUiMode sys
       // the swipe gesture. The overlays cannot be dismissed, so adding callback support will
       // allow users to restore the system ui and dismiss the overlays.
       // Not compatible with top/bottom overlays enabled.
-      enabledOverlays =
-          View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
-              | View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-              | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-              | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-              | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-              | View.SYSTEM_UI_FLAG_FULLSCREEN;
+      windowInsetsControllerCompat.setSystemBarsBehavior(
+          WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
+      windowInsetsControllerCompat.hide(WindowInsetsCompat.Type.systemBars());
+      WindowCompat.setDecorFitsSystemWindows(window, false);
     } else if (systemUiMode == PlatformChannel.SystemUiMode.EDGE_TO_EDGE
         && Build.VERSION.SDK_INT >= 29) {
```

Comment:
This version check can probably be removed now 

https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:core/core/src/main/java/androidx/core/view/WindowCompat.java;l=119-124?q=setDecorFitsSystemWindows

Reply:
Actually, it seems like there's a test that ensures this is not enabled for versions below this. @Piinks any insight on this?

Reply:
This is restricted on purpose, see https://github.com/flutter/flutter/issues/89774 for context.

Reply:
@Piinks For the comment below,  "SDK 29 and up will apply..." are these things set somewhere else?

I'm trying to understand where the differences are between the Flutter implementation and the internal Android implementation.

Reply:
Sorry for budding in here, but I did ask @Piinks about this yesterday, and it's my understanding that this comment is a bit misleading. The call made when the edge to edge mode is set in Flutter is just to lay out the app behind the system bars, [same for Android](https://developer.android.com/training/gestures/edge-to-edge#lay-out-in-full-screen). The remaining steps to achieve edge to edge mode like in Android (shown in the guide) in Flutter are done beyond a call to `setSystemChromeEnabledSystemUIMode`. Changing the color of the system bars, as mentioned in the comment, actually would have to be done by setting the overlay style via `setSystemChromeSystemUIOverlayStyle`. Please correct me if I'm wrong!